### PR TITLE
add Returns Google drive file contents lambda

### PR DESCRIPTION
**Pull request by snirkap**

### CODE REVIEW

File: backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import json
+import boto3
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+from googleapiclient.http import MediaIoBaseDownload
+import io
+
+def get_secret():
+    secret_name = "google_drive_service_account"  # Replace with the name of your secret
+    region_name = "us-east-1"   # Replace with your AWS region
```
Comment (entrypoint-community) [inline backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py]: get this from environment variable please

File: backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import json
+import boto3
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+from googleapiclient.http import MediaIoBaseDownload
+import io
+
+def get_secret():
+    secret_name = "google_drive_service_account"  # Replace with the name of your secret
```
Comment (entrypoint-community) [inline backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py]: get this environment variable please

File: backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    region_name = "us-east-1"   # Replace with your AWS region
+
+    # Create a Secrets Manager client
+    session = boto3.session.Session()
```
Comment (entrypoint-community) [inline backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py]: if you are not passing any arguments to the Session method, you might as well just use the `boto3.client` method to create the session.

File: backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    return json.loads(secret)
+
+def lambda_handler(event, context):
+    # Get service account credentials from AWS Secrets Manager
+    credentials_info = get_secret()
+    
+    # Load the service account credentials JSON data
+    credentials = service_account.Credentials.from_service_account_info(
+        credentials_info,
+        scopes=['https://www.googleapis.com/auth/drive.readonly']
```
Comment (entrypoint-community) [inline backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py]: please set this in a variable for future expansion (if one will be needed).

File: backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    file_id = event['file_id']
+    print("File ID: ", file_id)
+
+    # Determine the MIME type for export
+    mime_type = 'text/plain'  # Export as plain text
+    
+    # Export the file content
+    request = service.files().export_media(fileId=file_id, mimeType=mime_type)
+    file_content = io.BytesIO()
+    downloader = MediaIoBaseDownload(file_content, request)
```
Comment (entrypoint-community) [inline backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py:49]: just a small FYI - in a lambda environment, you can only save files to the /tmp directory, so it's good that you're not saving it and keeping it in memory.

File: backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    done = False
+    while done is False:
+        status, done = downloader.next_chunk()
+    file_content.seek(0)
+    
+    # Process the file content
+    contacts_str = file_content.read().decode('utf-8').lstrip('\ufeff')
```
Comment (entrypoint-community) [inline backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py:56]: can you explain this `lstrip('\ufeff')` please?

File: backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    print("Contacts string: ", contacts_str)
+    
+    # Create the JSON payload
+    payload = {
+        'contacts': contacts_str
+    }
+    
+    return {
+        'statusCode': 200,
+        'body': json.dumps(payload)
```
Comment (entrypoint-community) [inline backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py:71]: CORS

Comment (snirkap) [reply backend/lambda_functions/Returns_Google_drive_file_contents/Returns_Google_drive_file_contents_lambda.py:71]: i updated the lambda

**Review by entrypoint-community — APPROVED** [review]

**Pull request merged**